datafusion/physical-expr/src/scalar_function.rs

```diff
@@ -144,24 +144,11 @@ impl PhysicalExpr for ScalarFunctionExpr {
     fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
         // evaluate the arguments, if there are no arguments we'll instead pass in a null array
         // indicating the batch size (as a convention)
-        let inputs = match (
-            self.args.is_empty(),
-            self.name.parse::<BuiltinScalarFunction>(),
-        ) {
-            // MakeArray support zero argument but has the different behavior from the array with one null.
-            (true, Ok(scalar_fun))
-                if scalar_fun
-                    .signature()
-                    .type_signature
-                    .supports_zero_argument() =>
-            {
-                vec![ColumnarValue::create_null_array(batch.num_rows())]
-            }
+        let inputs = match self.args.is_empty() {
             // If the function supports zero argument, we pass in a null array indicating the batch size.
             // This is for user-defined functions.
-            (true, Err(_))
-                if self.supports_zero_argument && self.name != "make_array" =>
-            {
+            // MakeArray support zero argument but has the different behavior from the array with one null.
```

Comment:
This is a strange special case (that might be left over). @jayzhan211  or @Weijun-H  do you remember if this is still needed?

Reply:
I think we should even return empty vec instead of null array here

Reply:
File #10193 to fix this